lambda/writeFaq/index.js

```diff
@@ -48,28 +37,28 @@ exports.handler = async (event, context) => {
 async function updateItem(obj, context) {
   const { faq } = obj;
 
-  let updateParams = {
-    Key: {
-      pk: { S: 'faq' },
-      sk: { S: 'faq' }
-    },
-    UpdateExpression: 'set',
-    ExpressionAttributeValues: {},
-    ReturnValues: 'ALL_NEW',
-    TableName: TABLE_NAME,
-    ConditionExpression: 'attribute_exists(pk) AND attribute_exists(sk)'
-  };
-
-  if ('faq' in obj) {
-    updateParams.UpdateExpression += ' #text = :faq,';
-    updateParams.ExpressionAttributeValues[':faq'] = { S: faq };
-    updateParams.ExpressionAttributeNames = { '#text': 'text' };
-  }
-
-  updateParams.UpdateExpression = updateParams.UpdateExpression.slice(0, -1);
-  logger.debug('Updating FAQ:', updateParams);
-
   try {
+    if (!obj.faq){
+      throw new Error('FAQ property is missing in the input object');
+    }
+    let updateParams = {
+      Key: {
+        pk: { S: 'faq' },
+        sk: { S: 'faq' }
+      },
+      ExpressionAttributeNames: { '#text': 'text' },
+      ExpressionAttributeValues: {},
+      UpdateExpression: 'set',
+      ReturnValues: 'ALL_NEW',
+      TableName: TABLE_NAME,
+      ConditionExpression: 'attribute_exists(pk)'
+    };
+    if (obj?.faq) {
+      updateParams.UpdateExpression += ' #text = :faq,';
+      updateParams.ExpressionAttributeValues[':faq'] = { S: obj.faq };
+    }
+    updateParams.UpdateExpression = updateParams.UpdateExpression.slice(0, -1);
+    logger.debug('Updating FAQ:', updateParams);
```

Comment:
I would refactor as per my last comment:
```js
if (obj?.faq) {
   let updateParams = {
   ...
   // create and insert PUT item
   ...
   return sendResponse(200, AWS.DynamoDB.Converter.unmarshall(Attributes), context);
} else {
   throw new Error("FAQ property is missing in the input object");
}
```
It's 1 less `if` statement to achieve the same effect. Either way, on L41 ensure you are using optional chaining `obj?.faq` as checking for `obj.faq` will break in the event of a bad request object (`obj` has no property `faq`).

Reply:
Also curious if we will ever need the option to clear the faq:

Incoming request body:
```json
{
   faq: ''
}
```
As `obj.faq` is falsy in this case - probably not something that needs to be addressed right now.

Reply:
Addressed in : [commit](https://github.com/bcgov/parks-reso-api/pull/342/commits/ffbe60fb51c9bf3299ab351f08c3e5f476b42000)